app/components/Modal.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
+import Modal from "react-native-modal";
+import { darkBlue } from '../styles/Colours';
+
+export default class ModalComponent extends Component {
+
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			isModalVisible: props.isModalVisible,
+			isSimpleClose: props.isSimpleClose,
+			onClose: props.onClose
+		};
+
+		this.openModal = this.openModal.bind(this);
+		this.closeModal = this.closeModal.bind(this);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		const { isModalVisible } = nextProps;
+		if (isModalVisible !== this.state.isModalVisible) {
+			this.setState({ isModalVisible });
+		}
+	}
+
+	openModal() {
+		this.setState({ isModalVisible: true });
+	}
+
+	closeModal() {
+		if (this.state.isSimpleClose) this.setState({ isModalVisible: false });
+		else this.state.onClose();
+	}
+
+	render() {
+		return (
+			<Modal isVisible={ this.state.isModalVisible }>
+				<View style={ styles.modalContent }>
+					<Text style={ styles.title }>Success!</Text>
+					<Text style={ styles.body }>Your account has been created!  Login now to access the app.</Text>
+					<TouchableOpacity
+						style={ styles.button }
+						onPress={ this.closeModal }
+					>
+						<Text style={ styles.buttonText }>LOGIN</Text>
+					</TouchableOpacity>
+				</View>
+			</Modal>
+		);
+	}
+}
+
+ModalComponent.propTypes = {
+	isModalVisible: PropTypes.bool,
+	isSimpleClose: PropTypes.bool,
+	onClose: PropTypes.func
+};
+
+ModalComponent.defaultProps = {
+	isModalVisible: false,
+	isSimpleClose: false,
+	onClose: () => {}
+};
+
+const styles = StyleSheet.create({
```

Comment:
Should we leave styles in the same file as the component or move them to a different file? Is there a best practice?

Reply:
Hmm...  if they're small enough, they can be placed within the same file, but in this code, yea it doesn't make sense.  So, I've moved it to a separate file.